src/KafkaFlow.UnitTests/Consumer/WorkerPoolFeederTests.cs

```diff
@@ -43,13 +43,13 @@ public async Task StopAsync_WithoutStarting_Return()
         public async Task StopAsync_BlockedOnConsume_MustStop()
         {
             // Arrange
-            var consumeResult = new ConsumeResult<byte[], byte[]>();
-
             this.consumerMock
                 .Setup(x => x.ConsumeAsync(It.IsAny<CancellationToken>()))
-                .Returns(
-                    new ValueTask<ConsumeResult<byte[], byte[]>>(
-                        Task.Delay(Timeout.Infinite).ContinueWith(_ => consumeResult)));
+                .Returns(async () =>
+                {
+                    await Task.Delay(Timeout.Infinite);
+                    return default; // Never reached
+                });
```

Comment:
I can revert this file, there are no real changes, I had made a lot of changes here before realising the tests should remain the same but the implementation should change (although again, it could be better to remove this test and simplify the implementation as cooperative cancellation is probably good enough)